uui-core/src/types/pickers.ts

```diff
@@ -70,10 +82,32 @@ export type PickerBaseOptions<TItem, TId> = {
     /** Gets options for each row. Allow to make rows non-selectable, as well as many other tweaks. */
     getRowOptions?: (item: TItem, index: number) => DataRowOptions<TItem, TId>;
 
-    /** Overrides the default 'no records found' banner.
-     * The 'search' callback parameter allows to distinguish cases when there's no records at all, and when current search doesn't find anything.  */
+    /**
+     * @deprecated in favor of `renderEmpty` method.
+     * Overrides the default 'no records found' banner.
+     * The 'search' callback parameter allows to distinguish cases when there's no records at all, and when current search doesn't find anything.
+     * */
     renderNotFound?: (props: { search: string; onClose: () => void }) => ReactNode;
 
+    /**
+     * Overrides the rendering of PickerInput body content when it is empty for various reasons.
+     * If provided, the override should support messages for both 'not-found-records' and 'less-than-min-chars-to-search' reasons.
+     * If not provided, the `renderNotFound` method is used for the 'not-found-records' reason.
+     * @param props
+     * @param props.reason - Specifies the reason why the PickerInput body is empty:
+     *   - 'not-found-records': No data was found.
```

Comment:
There are 3 cases when there are no records in the Picker Body:

1. There are no items at all, e.g. empty array is passed
2. Search was run, but returned no elements
3. There are maxCharsToSearch set, and not enough chars entered to trigger search

1 and 2 can be distinguished with search != ''. However, 2 and 3 can also be distinguished with search.length < minCharsToSearch.

So I feel like this API is not consistent - we have 2 reasons in `reason`, instead of 3. And there's no clear logic why we choose this set.

Also, 'found' in 'not-found-records' imply that we were searching something, while in case 1 we did not.

To make API consistent, I'd either 
A) add a third reason, e.g. 'no-data'
B) remove the reason prop altogether. We can add `isSearchTooShort: boolean` to avoid adding using complex `search.length < minCharsToSearch`.

While I voted for A before, now I'm more into B. Because who knows what other reasons we got later? It can be hard to classify and name them. For example, if we'll opt to handle errors, should we distinguish errors with and without search? E.e. should we have separate "search-error" and "load-error" reasons?